domain/permission/state/shared.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"context"
+
+	"github.com/canonical/sqlair"
+	"github.com/juju/errors"
+
+	corepermission "github.com/juju/juju/core/permission"
+	permissionerrors "github.com/juju/juju/domain/permission/errors"
+	internaldatabase "github.com/juju/juju/internal/database"
+)
+
+// SharedState holds the shared state for the permission domain. Methods
+// which are shared between multiple domains.
+type SharedState struct {
+	statementBase internaldatabase.StatementBase
+}
+
+// NewSharedState creates a new SharedState.
+func NewSharedState(base internaldatabase.StatementBase) *SharedState {
```

Comment:
s/base/statementBase/

To help ensure no precendent is started, avoiding confusing with  base := corebase.Base`